front/pages/super-user/index.tsx

```diff
@@ -0,0 +1,86 @@
+import { GetServerSideProps, InferGetServerSidePropsType } from "next";
+import React from "react";
+
+import SuperUserNavbar from "@app/components/super-user/SuperUserNavbar";
+import { getSession, getUserFromSession } from "@app/lib/auth";
+import { useSuperUserWorkspaces } from "@app/lib/swr";
+import { UserType } from "@app/types/user";
+
+export const getServerSideProps: GetServerSideProps<{
+  user: UserType;
+}> = async (context) => {
+  const session = await getSession(context.req, context.res);
+  const user = await getUserFromSession(session);
+
+  if (!user) {
+    return {
+      redirect: {
+        destination: "/login",
+        permanent: false,
+      },
+    };
+  }
+
+  if (!user.isDustSuperUser) {
```

Comment:
No you should use auth here 

auth.isDustSuperUser

Reply:
I don't get it. What auth ? I don't have a workspace here, so I don't have an auth object. I just need to check whether a user is a super user.
Should I create a `isDustSuperUser` helper function in `auth.ts` that returns `user.isDustSuperUser` ?

Reply:
I understand this point for endpoints / routes that act on a specific workspace, but here I am listing all workspaces so I don't have a specific one to create an auth for

Reply:
Ok I see your point. So ideally, we want `Authenticator.dustSuperUser`, with an optional `workspace` param that returns an authenticator + a `isDustSuperUser` method ?

Reply:
`fromSession` so the `workspaceId` becomes optional, and add an optional `superUser` bool param ?

Reply:
Let’s stick with a User based authentication for now to keep things simple and easy as a v0 👍

I think you can use role admin when returning workspaces in poke so that we don’t introduce a new role if we don’t really need it for anything other than respecting a type signature ?

Reply:
The workspace less Auth support requires a bit of design doc work I think so that it’s as future proof as possible. I propose we handle it in future PR and keep things simple for v0 ?

Reply:
> I think you can use role admin when returning workspaces in poke so that we don’t introduce a new role if we don’t really need it for anything other than respecting a type signature ?

Sounds good

> The workspace less Auth support requires a bit of design doc work I think so that it’s as future proof as possible. I propose we handle it in future PR and keep things simple for v0 ?

aligned